src/business/dal/adapters/adapters.factory.ts

```diff
@@ -0,0 +1,22 @@
+import ApiAdapter from '../api/api.adapter';
+import IDataAdapter from '../data-adapter.interface';
+import FirebaseAdapter from '../firebase/firebase.adapter';
+import { Entity } from './entities.list';
+
+export enum AdapterType {
+  API,
+  FIREBASE,
+}
+
+export default class AdaptersFactory {
+  public createAdapter(type: AdapterType, entityKey: Entity): IDataAdapter {
+    switch (type) {
+      case AdapterType.API:
+        return new ApiAdapter(entityKey);
+      case AdapterType.FIREBASE:
+        return new FirebaseAdapter(entityKey);
+      default:
+        throw new Error(`This type ${type} of adapter does not exist`);
```

Comment:
### ☝ Suggestion: Use prevent function here

For example:

function generateSwitchSyntaxError(prop: never){
  throw new Error(`The property ${prop} isn't supported`);
}

switch (type) {
  case AdapterType.API:
    return new ApiAdapter(entityKey);
  case AdapterType.FIREBASE:
    return new FirebaseAdapter(entityKey);
  default:
    generateSwitchSyntaxError(type);
}

Reply:
If I do this, I have problems with types and I don't understand why we should use prevent function here for the error

![image](https://github.com/QoreCode/party-peyments/assets/9303844/34f4e3f7-0ecf-4150-b83f-23c4cd868c41)